### DISCUSSION TITLE
Problem with command_file

### DISCUSSION THREAD

**Discussion by jd-potato** [nulldriver/cf-cli-resource#105] [Q&A]
Hi there, by the way thanks for this project.

I try to use this option command_file to load a file at the execution of my pipeline but I always get the same error (even when I try with your sample code) : 
[ERROR] invalid payload (can not find command_file: cf_command/params.yml
It looks like it cannot find the file or the option command_file doesnt seems to work... 

I intercepted the container to see inside if the file is there, and there is the file into the good directory. So the file exists even tought it doesnt want to load the file with the command_file option. 

I tried with my personal code and with your exemple here : 
https://github.com/nulldriver/cf-cli-resource#command-via-file-input
and I still get this error no matter what the file contains.

Do you know what is the problem please ?

Thanks a lot

**Comment by patrickcrocker** [comment]
Greetings!  Happy to help!  Would you be able to share your pipeline with me to help you?

If you'd like, you can contact me directly @nulldriver on cloudfoundry.slack.com

**Reply by jd-potato** [reply]
Sorry I can't access to your slack channel...

**Reply by jd-potato** [reply]
do you have a full working exemple using command_file ?

**Reply by patrickcrocker** [reply]
Here is a [full example pipeline](/nulldriver/cf-cli-resource/tree/main/examples/command_file) and you can check it out in [action here](https://ci.nulldriver.com/teams/examples/pipelines/command_file).

Please take a look and let me know if this helps!

**Reply by jd-potato** [reply]
---
jobs:
  - name: command_file-example
    public: true
    plan:
      - task: create-org-and-space
        config:
          platform: linux
          image_resource:
            type: registry-image
            source:
              repository: busybox
          outputs:
            - name: task-output
          run:
            path: sh
            args:
              - -excl
              - |-
                cat > task-output/commands.yml <<EOF
                commands:
                [...some commands here ...]
                EOF
      - put: cloud-foundry
        params:
          command_file: task-output/commands.yml

resources: [...]

(sorry hypens above in the code changed for dot when I click on send comment...)

Thanks a lot this is very helpfull, I understand how to use it now. 
Unfortunately, I still have the same issue, wich is 
[ERROR] INVALID  PAYLOAD (can not find command_file: task-output/commands.yml
I tried and adapt to my situation the same exemple in your code, exemple above 

I intercepted the container and the file task-output/commands.yml exists, so the task create correctly the file with all the its content. But I got the ERROR invalid payload at the PUT step...

and i got the same error if I try to get the file from a repos instead of create it directly in the pipeline. 

i'm wondering Is command_file compatible with concourse version 6.7.8 ?

**Reply by patrickcrocker** [reply]
Strange, and I don't think your version of Concouse would matter...

It may help me if you can provide me your full pipeline yml.  I've created a private repo and invited you to it.  It would be great if you could upload your sample pipeline to that repo (please use placeholders for any sensitive settings) so that I can help you further.

After we resolve this, we can delete the private repo.  Let me know!

**Reply by jd-potato** [reply]
It is really kind of you, I really appreciate your help. Unfortunately, I can't share any part of my pipeline, I am not allowed to share anything. But we found another option instead of using command_file and it is working as we want.
Maybe someday we will find what was our problem with this feature, who knows ?
Have a nice day
Thanks again Patrick for all your involvement
